enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,502 @@
+    ...
+    // External contains settings specific to the generic External infrastructure provider.
+    // +optional
+    External *ExternalPlatformStatus `json:"external,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints
+
+#### Implementation phases
+
+Splitting the project into phases would be natural to make the implementation process smoother.
+A reader can find the proposed phase breakdown in [OCPPLAN-9429](https://issues.redhat.com/browse/OCPPLAN-9429).
+
+This document intends to describe the initial phases of this project. The proposed initial course of action:
+
+1. Update "openshift/api" with adding "External" PlatformType
+2. Ensure that all Red Hat operators tolerate the "External" platform and treat the same as the "None" platform
+
+Next phase out of the scope for this EP:
+
+3. Modify operators for doing specific things when seeing the "External" platform.
+   It was described briefly in the [Affected Components](#affected-components) section.
+   However, this should be addressed in separate EPs on a per-component basis.
+
+#### Future work
+
+There are several topics in this area that would be wise to defer for upcoming phases, namely:
+
+1. Define missing capabilities and their concrete behaviour, for example, add a "capability" for machine-api
+2. Precisely define the reaction of the operators listed below for the "External" platform type
+3. Define and document concrete mechanism for supplement cluster with provider-specific components during installation time (CCM, MAPI controller, DNS controller)
+
+### Risks and Mitigations
+
+#### Support burden
+There is concern that *random* customers will use this feature out of context
+and create a support burden. However, using such platform type in conjunction with [capabilites selection](https://github.com/openshift/enhancements/blob/master/enhancements/installer/component-selection.md)
+could give us a clear signal about how to properly triage 3d party created replacements for Red Hat components.
+
+### Drawbacks
+
+Current approach with having statically defined platform list is already there, quite transparent and battle-hardened.
+Changing it by adding the new, way less specific platform type would mean a significant step away from current design patterns in this area.
+
+Also, future support strategy is not completely clear due to we have plans around enablement of 3d party platform-specific components, which
+codebase would be mostly out of our control.
+
+## Design Details
+
+### Open Questions
+
+- Should we explicitly communicate that the "External" platform is something that we do not support yet?
+- Should we gate the "External" platform addition behind the feature gate by generating separate CRD for TPNU clusters?
+  - Do we already have all the necessary mechanisms for gating such API changes behind the feature gate?
+- Should we invest in preparing workflows that will perform UPI with the "None"/"External" platform types installation
```

Comment:
reminder that assisted-installer is generally a good alternative to UPI and should be a good fit for installing with the "External" platform type.

Reply:
+1, i am curious to know more about install flexibility will interact with agent install scenarios.